File: nmLabReport/lab32.c
#include <stdio.h>

// Define the differential equation dy/dx = f(x, y)
double f(double x, double y) {
    return (x + y); // Example: dy/dx = x + y
}

// Function to implement the Fourth Order Runge-Kutta Method
void rungeKutta(double x0, double y0, double x, double h) {
    int n = (int)((x - x0) / h); // Number of iterations
    double k1, k2, k3, k4, y = y0;

    printf("x0\ty0\n");
    printf("%lf\t%lf\n", x0, y0);

    for (int i = 1; i <= n; i++) {
        k1 = h * f(x0, y);
        k2 = h * f(x0 + 0.5 * h, y + 0.5 * k1);
        k3 = h * f(x0 + 0.5 * h, y + 0.5 * k2);
        k4 = h * f(x0 + h, y + k3);

        y = y + (1.0 / 6.0) * (k1 + 2 * k2 + 2 * k3 + k4);
        x0 = x0 + h;

        printf("%lf\t%lf\n", x0, y);
    }
}

int main() {
    double x0 = 0, y0 = 1; // Initial conditions
    double x = 2; // Point at which we need the solution
    double h = 0.2; // Step size

    rungeKutta(x0, y0, x, h);

    return 0;
}

File: nmLabReport/lab32.py
def f(x, y):
    return x + y  # Example: dy/dx = x + y

def runge_kutta(x0, y0, x, h):
    n = int((x - x0) / h)
    y = y0

    print(f"x0 = {x0:.6f}, y0 = {y0:.6f}")

    for i in range(1, n + 1):
        k1 = h * f(x0, y)
        k2 = h * f(x0 + 0.5 * h, y + 0.5 * k1)
        k3 = h * f(x0 + 0.5 * h, y + 0.5 * k2)
        k4 = h * f(x0 + h, y + k3)

        y = y + (1.0 / 6.0) * (k1 + 2 * k2 + 2 * k3 + k4)
        x0 = x0 + h

        print(f"x{i} = {x0:.6f}, y{i} = {y:.6f}")

def main():
    x0 = 0  # Initial conditions
    y0 = 1  # Initial conditions
    x = 2  # Point at which we need the solution
    h = 0.2  # Step size

    runge_kutta(x0, y0, x, h)

if __name__ == "__main__":
    main()